.travis.yml

```diff
@@ -37,7 +37,7 @@ after_success:
   - |
         if [ "$COMMAND" == "end-to-end-test" ]; then
             nyc report
-            nyc report --reporter=lcov | codecov
+            nyc report --reporter=lcov > coverage.lcov && codecov
```

Comment:
We need to know what the root cause of the problem is. As you can see above on line 36 `codecov` can easily read the lcov information from its stdin. It doesn't need to read that information from a file. And if you inspect the CI build logs for the various test jobs you can see that for the unit-test job it can happily read this information. Also codecov was able to happily read the lcov information from stdin prior to PR #135. 

The problem lies elsewhere and until we know where I don't feel comfortable accepting this patch. Things should either work or not work. Instead we have the case of "was working" and "stopped working without any related changes".

Reply:
This is the log from PR#136 which got a right coverage result.
```shell
==> Generating gcov reports (skip via --disable=gcov)
    $ find /home/travis/build/weldr/welder-web -type f -name '*.gcno'  -exec gcov  {} +
==> Scanning for reports
    + /home/travis/build/weldr/welder-web/coverage/lcov.info
==> Uploading reports
    Success!
```
Here's the correct log:
```shell
==> Generating gcov reports (skip via --disable=gcov)
    $ find /home/travis/build/weldr/welder-web -type f -name '*.gcno'  -exec gcov  {} +
==> Scanning for reports
    + /home/travis/build/weldr/welder-web/coverage.lcov
    + /home/travis/build/weldr/welder-web/coverage/lcov.info
==> Uploading reports
```
The only thing codecov did is scanning file which match the pattern and upload the matched result file into server. The code can be found from https://github.com/codecov/codecov-node/blob/master/lib/codecov.js
So even the stdin was passed to codecov, it'll ignore it and does scanning result file according to pattern. From the codecov.js, it only accepts the following types of files:
```shell
'*coverage.*' + 'nosetests.xml' + 'jacoco*.xml' + 'clover.xml' + 'report.xml' + 'cobertura.xml' +'luacov.report.out' + 'lcov.info' +'*.lcov' +'gcov.info' +'*.gcov' +'*.lst'
```
So the .json results has to be converted into one of above file type. Then the codecov will find it and upload it to server.

Reply:
For the Line 36, if removing `cat ./coverage/lcov.info | `, it still works. 
The only thing codecov did is scanning file which match the pattern and upload the matched result file into server. The codecov source code can be found from https://github.com/codecov/codecov-node/blob/master/lib/codecov.js
So even the stdin was passed to codecov, codecov'll ignore it and does scanning result file according to pattern. From the codecov.js, it only accepts the following types of files:
```shell
'*coverage.*' + 'nosetests.xml' + 'jacoco*.xml' + 'clover.xml' + 'report.xml' + 'cobertura.xml' +'luacov.report.out' + 'lcov.info' +'*.lcov' +'gcov.info' +'*.gcov' +'*.lst'
```